azurerm_container_app probe failure threshold upper bound is too low

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.14.7

### AzureRM Provider Version

4.64.0

### Affected Resource(s)/Data Source(s)

azurerm_container_app

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "repro" {
  name     = "repro-rg"
  location = "eastus"
}

resource "azurerm_container_app_environment" "repro" {
  name                = "repro-cae"
  location            = azurerm_resource_group.repro.location
  resource_group_name = azurerm_resource_group.repro.name
}

resource "azurerm_container_app" "repro" {
  name                         = "repro-app"
  container_app_environment_id = azurerm_container_app_environment.repro.id
  resource_group_name          = azurerm_resource_group.repro.name
  revision_mode                = "Single"

  template {
    container {
      name   = "repro-container"
      image  = "mcr.microsoft.com/azuredocs/containerapps-helloworld:latest"
      cpu    = 0.25
      memory = "0.5Gi"

      readiness_probe {
        failure_count_threshold = 48
        interval_seconds        = 5
        path                    = "/healthz"
        port                    = 8080
        success_count_threshold = 1
        timeout                 = 5
        transport               = "HTTP"
      }

      startup_probe {
        failure_count_threshold = 240
        initial_delay           = 1
        interval_seconds        = 1
        path                    = "/healthz"
        port                    = 8080
        timeout                 = 3
        transport               = "HTTP"
      }
    }
  }
}
```

### Debug Output/Panic Output

```shell
2026-03-17T12:24:43.168Z [TRACE] Validating config for "azurerm_container_app.repro"
2026-03-17T12:24:43.168Z [TRACE] GRPCProvider: ValidateResourceConfig
2026-03-17T12:24:43.168Z [TRACE] GRPCProvider: GetProviderSchema
2026-03-17T12:24:43.168Z [TRACE] GRPCProvider: returning cached schema: EXTRA_VALUE_AT_END=registry.terraform.io/hashicorp/azurerm
2026-03-17T12:24:43.169Z [TRACE] provider.terraform-provider-azurerm_v4.64.0_x5.exe: Received request: tf_proto_version=5.10 tf_rpc=ValidateResourceTypeConfig tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_id=a62a8583-4c27-0a89-a146-2d87ae71c82a tf_resource_type=azurerm_container_app @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/tf5server/server.go:762 @module=sdk.proto timestamp=2026-03-17T12:24:43.169Z
2026-03-17T12:24:43.169Z [TRACE] provider.terraform-provider-azurerm_v4.64.0_x5.exe: Announced client capabilities: @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/tf5serverlogging/client_capabilities.go:24 tf_proto_version=5.10 tf_provider_addr=registry.terraform.io/hashicorp/azurerm @module=sdk.proto tf_client_capability_write_only_attributes_allowed=true tf_req_id=a62a8583-4c27-0a89-a146-2d87ae71c82a tf_resource_type=azurerm_container_app tf_rpc=ValidateResourceTypeConfig timestamp=2026-03-17T12:24:43.169Z
2026-03-17T12:24:43.169Z [TRACE] provider.terraform-provider-azurerm_v4.64.0_x5.exe: Sending request downstream: @module=sdk.proto tf_proto_version=5.10 tf_provider_addr=registry.terraform.io/hashicorp/azurerm @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:22 tf_req_id=a62a8583-4c27-0a89-a146-2d87ae71c82a tf_resource_type=azurerm_container_app tf_rpc=ValidateResourceTypeConfig timestamp=2026-03-17T12:24:43.169Z
2026-03-17T12:24:43.169Z [TRACE] provider.terraform-provider-azurerm_v4.64.0_x5.exe: calling downstream server: tf_mux_provider="*schema.GRPCProviderServer" tf_rpc=ValidateResourceTypeConfig @caller=github.com/hashicorp/terraform-plugin-mux@v0.21.0/internal/logging/mux.go:19 @module=sdk.mux timestamp=2026-03-17T12:24:43.169Z
2026-03-17T12:24:43.169Z [TRACE] provider.terraform-provider-azurerm_v4.64.0_x5.exe: Calling downstream: @module=sdk.helper_schema tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_id=a62a8583-4c27-0a89-a146-2d87ae71c82a tf_rpc=ValidateResourceTypeConfig tf_mux_provider="*schema.GRPCProviderServer" tf_resource_type=azurerm_container_app @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/grpc_provider.go:435 timestamp=2026-03-17T12:24:43.169Z
2026-03-17T12:24:43.169Z [TRACE] provider.terraform-provider-azurerm_v4.64.0_x5.exe: Called downstream: @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.1/helper/schema/grpc_provider.go:437 @module=sdk.helper_schema tf_mux_provider="*schema.GRPCProviderServer" tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_id=a62a8583-4c27-0a89-a146-2d87ae71c82a tf_resource_type=azurerm_container_app tf_rpc=ValidateResourceTypeConfig timestamp=2026-03-17T12:24:43.169Z
2026-03-17T12:24:43.169Z [TRACE] provider.terraform-provider-azurerm_v4.64.0_x5.exe: Received downstream response: tf_proto_version=5.10 tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_duration_ms=0 tf_rpc=ValidateResourceTypeConfig @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:61 diagnostic_error_count=2 diagnostic_warning_count=0 tf_req_id=a62a8583-4c27-0a89-a146-2d87ae71c82a tf_resource_type=azurerm_container_app @module=sdk.proto timestamp=2026-03-17T12:24:43.169Z
2026-03-17T12:24:43.169Z [ERROR] provider.terraform-provider-azurerm_v4.64.0_x5.exe: Response contains error diagnostic: diagnostic_attribute="AttributeName(\"template\").ElementKeyInt(0).AttributeName(\"container\").ElementKeyInt(0).AttributeName(\"startup_probe\").ElementKeyInt(0).AttributeName(\"failure_count_threshold\")" diagnostic_detail="" diagnostic_summary="expected template.0.container.0.startup_probe.0.failure_count_threshold to be in the range (1 - 30), got 240" tf_proto_version=5.10 tf_req_id=a62a8583-4c27-0a89-a146-2d87ae71c82a diagnostic_severity=ERROR tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_resource_type=azurerm_container_app tf_rpc=ValidateResourceTypeConfig @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto timestamp=2026-03-17T12:24:43.169Z
2026-03-17T12:24:43.169Z [ERROR] provider.terraform-provider-azurerm_v4.64.0_x5.exe: Response contains error diagnostic: diagnostic_severity=ERROR tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_req_id=a62a8583-4c27-0a89-a146-2d87ae71c82a tf_resource_type=azurerm_container_app tf_rpc=ValidateResourceTypeConfig @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/diag/diagnostics.go:58 @module=sdk.proto diagnostic_attribute="AttributeName(\"template\").ElementKeyInt(0).AttributeName(\"container\").ElementKeyInt(0).AttributeName(\"readiness_probe\").ElementKeyInt(0).AttributeName(\"failure_count_threshold\")" diagnostic_summary="expected template.0.container.0.readiness_probe.0.failure_count_threshold to be in the range (1 - 30), got 48" tf_proto_version=5.10 diagnostic_detail="" timestamp=2026-03-17T12:24:43.169Z
2026-03-17T12:24:43.169Z [TRACE] provider.terraform-provider-azurerm_v4.64.0_x5.exe: Served request: tf_proto_version=5.10 tf_provider_addr=registry.terraform.io/hashicorp/azurerm tf_resource_type=azurerm_container_app tf_rpc=ValidateResourceTypeConfig @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/tf5server/server.go:783 @module=sdk.proto tf_req_id=a62a8583-4c27-0a89-a146-2d87ae71c82a timestamp=2026-03-17T12:24:43.169Z
2026-03-17T12:24:43.169Z [ERROR] vertex "azurerm_container_app.repro" error: expected template.0.container.0.startup_probe.0.failure_count_threshold to be in the range (1 - 30), got 240
2026-03-17T12:24:43.170Z [ERROR] vertex "azurerm_container_app.repro" error: expected template.0.container.0.readiness_probe.0.failure_count_threshold to be in the range (1 - 30), got 48
2026-03-17T12:24:43.170Z [TRACE] vertex "azurerm_container_app.repro": visit complete, with errors
2026-03-17T12:24:43.170Z [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/hashicorp/azurerm\"] (close)" errored, so skipping
2026-03-17T12:24:43.170Z [TRACE] dag/walk: upstream of "root" errored, so skipping
```

### Expected Behaviour

The Azure Portal allows (and indeed defaults) to setting the **Failure Threshold** above what the AzureRM provider allows.

_Liveness Probe_
Provider: 10
Azure: 30
<img width="597" height="486" alt="Image" src="https://github.com/user-attachments/assets/3ba75318-72b2-4634-bd4b-70d86d303590" />

_Readiness Probe_
Provider: 30
Azure: 48
<img width="545" height="492" alt="Image" src="https://github.com/user-attachments/assets/7e346eae-509e-46fa-979f-304926f58703" />

_Startup Probe:
Provider: 30
Azure: 240
<img width="542" height="473" alt="Image" src="https://github.com/user-attachments/assets/9470a171-7a5a-4070-b8da-cc004867b5cd" />

### Actual Behaviour

Errors when attempting to use Azure supported values, e.g.:

* `Error: expected template.0.container.0.readiness_probe.0.failure_count_threshold to be in the range (1 - 30), got 48`
* `Error: expected template.0.container.0.startup_probe.0.failure_count_threshold to be in the range (1 - 30), got 240`

### Steps to Reproduce

1. `terraform plan`

### Important Factoids

_No response_

### References

I could not find Microsoft documentation that defines to maximum limits. However they are used in the documentation for container apps relating to health probes: https://learn.microsoft.com/en-us/azure/container-apps/health-probes

--- a/internal/services/containerapps/helpers/container_apps.go
+++ b/internal/services/containerapps/helpers/container_apps.go
@@ -2066,8 +2066,8 @@ func ContainerAppReadinessProbeSchema() *pluginsdk.Schema {
 					Type:         pluginsdk.TypeInt,
 					Optional:     true,
 					Default:      3,
-					ValidateFunc: validation.IntBetween(1, 30),
-					Description:  "The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `30`. Defaults to `3`.",
+					ValidateFunc: validation.IntBetween(1, 48),
+					Description:  "The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `48`. Defaults to `3`.",
 				},
 
 				"success_count_threshold": {
@@ -2153,7 +2153,7 @@ func ContainerAppReadinessProbeSchemaComputed() *pluginsdk.Schema {
 				"failure_count_threshold": {
 					Type:        pluginsdk.TypeInt,
 					Computed:    true,
-					Description: "The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `30`. Defaults to `3`.",
+					Description: "The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `48`. Defaults to `3`.",
 				},
 
 				"success_count_threshold": {
@@ -2636,8 +2636,8 @@ func ContainerAppStartupProbeSchema() *pluginsdk.Schema {
 					Type:         pluginsdk.TypeInt,
 					Optional:     true,
 					Default:      3,
-					ValidateFunc: validation.IntBetween(1, 30),
-					Description:  "The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `30`. Defaults to `3`.",
+					ValidateFunc: validation.IntBetween(1, 240),
+					Description:  "The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `240`. Defaults to `3`.",
 				},
 			},
 		},
@@ -2721,7 +2721,7 @@ func ContainerAppStartupProbeSchemaComputed() *pluginsdk.Schema {
 				"failure_count_threshold": {
 					Type:        pluginsdk.TypeInt,
 					Computed:    true,
-					Description: "The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `30`. Defaults to `3`.",
+					Description: "The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `240`. Defaults to `3`.",
 				},
 			},
 		},

--- a/website/docs/r/container_app.html.markdown
+++ b/website/docs/r/container_app.html.markdown
@@ -262,7 +262,7 @@ A `container` block supports the following:
 
 A `liveness_probe` block supports the following:
 
-* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `10`. Defaults to `3`.
+* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `30`. Defaults to `3`.
 
 * `header` - (Optional) A `header` block as detailed below.
 
@@ -304,7 +304,7 @@ An `env` block supports the following:
 
 A `readiness_probe` block supports the following:
 
-* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `30`. Defaults to `3`.
+* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `48`. Defaults to `3`.
 
 * `header` - (Optional) A `header` block as detailed below.
 
@@ -336,7 +336,7 @@ A `header` block supports the following:
 
 A `startup_probe` block supports the following:
 
-* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `30`. Defaults to `3`.
+* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `240`. Defaults to `3`.
 
 * `header` - (Optional) A `header` block as detailed below.
 

--- a/website/docs/r/container_app_job.html.markdown
+++ b/website/docs/r/container_app_job.html.markdown
@@ -208,7 +208,7 @@ A `liveness_probe` block supports the following:
 
 * `transport` - (Required) Type of probe. Possible values are `TCP`, `HTTP`, and `HTTPS`.
 
-* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `10`. Defaults to `3`.
+* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `30`. Defaults to `3`.
 
 * `header` - (Optional) A `header` block as detailed below.
 
@@ -238,7 +238,7 @@ A `readiness_probe` block supports the following:
 
 * `transport` - (Required) Type of probe. Possible values are `TCP`, `HTTP`, and `HTTPS`.
 
-* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `10`. Defaults to `3`.
+* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `48`. Defaults to `3`.
 
 * `header` - (Optional) A `header` block as detailed below.
 
@@ -270,7 +270,7 @@ A `startup_probe` block supports the following:
 
 * `transport` - (Required) Type of probe. Possible values are `TCP`, `HTTP`, and `HTTPS`.
 
-* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `10`. Defaults to `3`.
+* `failure_count_threshold` - (Optional) The number of consecutive failures required to consider this probe as failed. Possible values are between `1` and `240`. Defaults to `3`.
 
 * `header` - (Optional) A `header` block as detailed below.
 

